sigstore-java/src/main/java/dev/sigstore/bundle/BundleVerifierInternal.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2023 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.bundle;
+
+import com.google.api.FieldBehavior;
+import com.google.api.FieldBehaviorProto;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.MessageOrBuilder;
+import com.google.protobuf.util.JsonFormat;
+import dev.sigstore.proto.bundle.v1.Bundle;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** Implements Sigstore Bundle verification. */
+public class BundleVerifierInternal {
+  static final JsonFormat.Parser JSON_PARSER = JsonFormat.parser();
+
+  /**
+   * Parses Sigstore Bundle JSON into protobuf.
+   *
+   * @param bundleJson input JSON
+   * @return parsed bundle
+   * @throws IllegalArgumentException if the JSON is invalid
+   */
+  static Bundle.Builder parseSigstoreBundle(String bundleJson) {
+    var bundle = Bundle.newBuilder();
+    try {
+      JSON_PARSER.merge(bundleJson, bundle);
+    } catch (InvalidProtocolBufferException e) {
+      throw new IllegalArgumentException("Unable to parse Sigstore Bundle " + bundleJson, e);
+    }
+    return bundle;
+  }
+
+  /**
+   * Finds all the missing field in the given protobuf message.
+   *
+   * @param message input message
+   * @return list of all the missing fields, or empty list if there are none
+   */
+  static List<String> findMissingFields(MessageOrBuilder message) {
+    final List<java.lang.String> results = new ArrayList<>();
+    findMissingFields(message, "", results);
+    return results;
+  }
+
+  private static void findMissingFields(
+      MessageOrBuilder message, String prefix, List<String> results) {
```

Comment:
```suggestion
      MessageOrBuilder message, String prefix, List<String> missing) {
```
`missing` or something similar feels a bit more descriptive, but this is relatively minor.